middleware/password.recovery.js

```diff
@@ -0,0 +1,66 @@
+
+const nodemailer = require('nodemailer')
+const crypto = require('crypto')
+const dotenv = require('promise-dotenv').config()
+
+async function generatePasswordResetToken() {
+  const token = await crypto.randomBytes(20).toString('hex');
+  return token;
+}
+
+
+async function forgetPassword(req,res){
+    const {email} = req.body
+    const user = await user.findOne({email})
+    if(!user){
+        return res.status(404).json({success: false, message: 'User not found' })
+    }
+    const token = await generatePasswordResetToken()
+  user.passwordResetToken = token;
+  await user.save()
+
+  const transporter = nodemailer.createTransport({
+    auth: {
+        user: 'cijeoma559@gmail.com',
+        pass: process.env.NODEMAILER
+    },
+  });
+  const mailOption = {
+    from: 'cijeoma559@gmail.com',
+    to: email,
+    subject: 'Reset your password',
+    text: `Click the following link to reset your password: ${app-link}/reset-password/${token}`,
+  };
+
+  try{
```

Comment:
The trycatch may  not be necessary as the sendMail method takes a callback

`transporter.sendMail(mailOptions, (error, info) => {
    if (error) {
      console.error("Error sending email: " + error);
    } else {
      console.log("Email sent: " + info.response);
    }
  });`